homeassistant/components/zamg/sensor.py

```diff
@@ -241,36 +245,72 @@ def setup_platform(
     )
 
 
-class ZamgSensor(SensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the ZAMG sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is None:
+        coordinator.config_entry = entry
+
+    async_add_entities(
+        [
+            ZamgSensor(coordinator, entry.title, description)
+            for description in SENSOR_TYPES
+        ],
+        False,
+    )
+
+
+class ZamgSensor(CoordinatorEntity, SensorEntity):
     """Implementation of a ZAMG sensor."""
 
     entity_description: ZamgSensorEntityDescription
 
-    def __init__(self, probe, name, description: ZamgSensorEntityDescription):
+    def __init__(self, coordinator, name, description: ZamgSensorEntityDescription):
         """Initialize the sensor."""
+        super().__init__(coordinator)
         self.entity_description = description
-        self.probe = probe
+        self.coordinator = coordinator
         self._attr_name = f"{name} {description.key}"
+        self._attr_unique_id = f"{self.coordinator.data.station_id()}_{description.key}"
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, self.platform.config_entry.unique_id)},
+            # manufacturer=,
+            # model=MODEL,
+            name=self.coordinator.name,
+        )
 
     @property
     def native_value(self):
         """Return the state of the sensor."""
-        return self.probe.get_data(self.entity_description.key)
+        return self.coordinator.data.get(self.entity_description.key)
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
+        update_time = datetime.strptime(
+            f"{self.coordinator.data.get('update_date')} {self.coordinator.data.get('update_time')}",
+            "%d-%m-%Y %H:%M",
+        )
         return {
             ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_STATION: self.probe.get_data("station_name"),
-            ATTR_UPDATED: self.probe.last_update.isoformat(),
+            ATTR_STATION: self.coordinator.data.get("station_name"),
+            CONF_STATION_ID: self.coordinator.config_entry.unique_id,
+            ATTR_UPDATED: update_time.isoformat(),
         }
 
     def update(self):
         """Delegate update to probe."""
-        self.probe.update()
+        self.coordinator.data.update()
```

Comment:
The coordinator already has an update interval, I don't think it's the idea to update it from here as well as then the interval is not controlled from a single place.